### ISSUE TITLE
[Feature] Add distribution type of "range" for grid method

### ISSUE THREAD

**Issue by XiaomoWu** [wandb/client#2665]
Currently, if I want to try out 5 random seeds, then I have to specify in the config.yaml file like this (suppose I'm using grid search):

```yaml
method: grid

seed:
    values: [0,1,2,3,4]
```
However, the key `values` become useless when the number of trials is large. For example, if I want to try every integer random seed from 0 to 1000, it just doesn't make too much sense to write out all the candidates.

Is it possible to add something like
```yaml
seed:
    distribution: range
    min: 0
    max: 1000
    step: 1
```

I apologize if it's already been implemented but I missed it.
Thank you!

**Comment by vanpelt** [comment]
Hey @XiaomoWu you should be able to do this today with:

```
seed:
  distribution: int_uniform
  min: 0
  max: 1000
```

You'll need to change the method to "random" instead of grid but the system should explore all values between 0 and 1000 in that case.  I'll keep this open as we may want to add support to grid search supporting bounded distributions like this.

**Comment by XiaomoWu** [comment]
> Hey @XiaomoWu you should be able to do this today with:
> 
> ```
> seed:
>   distribution: int_uniform
>   min: 0
>   max: 1000
> ```
> 
> You'll need to change the method to "random" instead of grid but the system should explore all values between 0 and 1000 in that case. I'll keep this open as we may want to add support to grid search supporting bounded distributions like this.

Thanks for the response! Just a few things to confirm:

- If I use "random", then I have to manually set the total count of runs or otherwise, wandb will run forever?

- Will wandb resample the same set of hparams before it exhausts all the possible combinations, i.e., does it sample with replacement (suppose I have many other hparams)?

**Comment by ash0ts** [comment]
Hello @XiaomoWu,

Sorry for the late response on this. We released https://github.com/wandb/sweeps/pull/49 which supports bounded distributions for grid methods a month ago. Can you let us know if this solution matches your desired use case?

**Comment by XiaomoWu** [comment]
> Hello @XiaomoWu,
> 
> Sorry for the late response on this. We released [wandb/sweeps#49](https://github.com/wandb/sweeps/pull/49) which supports bounded distributions for grid methods a month ago. Can you let us know if this solution matches your desired use case?

Thanks for your work! I'm closing the issue now.
